### PR TITLE
feat(reports): reports page foundation — route, nav, chart component, period selector

### LINKED ISSUE #313: Reports page foundation: route, nav, charts, and period selector
Part of #312. The foundation every reporting chart builds on. Adds the Reports page, the charting library, and the shared period control — but no individual charts yet (those are separate sub-issues).

## Scope

- Add the official ShadCN `chart` component (`src/components/ui/chart.tsx`) and its Recharts dependency via `pnpm dlx shadcn@latest add chart` (or the project's established ShadCN-add flow). Recharts arrives as the ShadCN chart engine — do not add a different charting library.
- Create a `/reports` route at `src/app/(app)/reports/page.tsx` rendering a page shell (heading + responsive grid of card slots for the chart widgets).
- Add a **Reports** entry to `AppShellNav` (`src/components/app-shell/AppShellNav.tsx`), with its label in the co-located `copy.ts`, and update the nav story/spec.
- Build a shared **period / date-range selector** (e.g. "Last 3 / 6 / 12 months / All time") as a presentational component that exposes the selected range; the reports page owns the state and passes the resolved range down to chart widgets (added in later sub-issues). Store its user-facing strings in a co-located copy file.
- Add a Storybook story and spec for the period selector and the page shell.

## Acceptance criteria

- [ ] `pnpm tsc`, `pnpm lint`, `pnpm test`, `pnpm build` all pass
- [ ] Navigating to `/reports` renders the page shell with the period selector and an empty/placeholder chart grid
- [ ] A **Reports** link appears in the app nav and routes to `/reports`
- [ ] ShadCN `chart` component is available for later sub-issues to import
- [ ] Period selector is a tested presentational component; changing it updates the page's selected range (verified in a spec)
- [ ] User-facing strings live in co-located `copy.ts` files, asserted against in tests

## Notes

- This is the dependency for the four chart sub-issues — they should stack on this PR (or wait for it to merge).
- No Firebase schema changes.

---
*Created by Claude Opus 4.8*

### PR DESCRIPTION
**Pull request by rmartz**
The foundation every Reporting chart builds on — the `/reports` route, the charting library, the nav entry, and the shared period control. No individual charts yet (those are the four dependent sub-issues #314–#317, which can stack on this branch).

## Key decisions

- **ShadCN `chart` + Recharts** added via `pnpm dlx shadcn@latest add chart` (`src/components/ui/chart.tsx`, `recharts` pinned `3.8.0`). The `chart` add also tried to re-generate the shared `card.tsx` to a newer base-nova style (dropping its exported prop interfaces, changing `CardTitle` from `<h2>` to `<div>`) — I **reverted `card.tsx`**, since that's an unrelated breaking change to a widely-used component and `chart.tsx` doesn't depend on it.
- **Period value-set is structural** — `REPORT_PERIODS` (`as const`) + a derived `ReportPeriod` union, per the value-set convention (assigns cleanly to a future `?period=` query param, no cast). A pure `resolvePeriodRange(period, now)` resolves it to a `[start, end]` range (all-time → unbounded start); it takes `now` as a parameter for deterministic tests, and is exported for the chart widgets to consume.
- **Presentational split** — `ReportsPeriodSelector` and `ReportsView` are presentational; the `/reports` route owns the selected-period state and passes it down. User-facing strings live in `src/components/reports/copy.ts`.

## Acceptance Criteria
- [x] `pnpm tsc`, `pnpm lint`, `pnpm test`, `pnpm build` all pass
- [x] Navigating to `/reports` renders the page shell with the period selector and an empty/placeholder chart grid
- [x] A **Reports** link appears in the app nav and routes to `/reports`
- [x] ShadCN `chart` component is available for later sub-issues to import
- [x] Period selector is a tested presentational component; changing it updates the page's selected range (verified in a spec)
- [x] User-facing strings live in co-located `copy.ts` files, asserted against in tests

## Tests
13 reports tests (range resolver, selector, page-shell + stateful-harness) + a nav spec for the Reports link. Full suite green (1275 tests); production build lists `/reports`.

Closes #313

---
*Created by Claude Opus 4.8*

### CODE REVIEW

**Review by rmartz — COMMENTED** [review]
## Requested changes

- **`resolvePeriodRange` month-end overflow in `period.ts`**: `setMonth` subtraction silently rolls over when the current day doesn't exist in the target month. For example, if today is March 31 and you subtract 1 month, `setMonth(1)` (February) has 28 or 29 days, so JavaScript normalizes it to March 2 or 3. For a user running a "last 3 months" report on July 31, the start date becomes May 1 (skipping all of April) instead of April 30. The fix is to reset the day to 1 before subtracting months, or — more natural for financial reporting — define period start as the first of the month N months ago: `new Date(now.getFullYear(), now.getMonth() - months, 1)`. The period.spec.ts tests should also cover a month-end reference date (e.g. the 31st) so this edge case is verified going forward.

## Verdict

One correctness bug in the period resolver — the foundational `resolvePeriodRange` function needs the overflow fix before the chart sub-issues (#314–317) stack on this.

---

_Claude Sonnet 4.6_

**Comment by rmartz** [comment]
Fixed the month-end day overflow in `resolvePeriodRange`:

- **`period.ts`**: replaced `setMonth` subtraction with `new Date(now.getFullYear(), now.getMonth() - months, 1)` so period starts always land on the first of the month N months prior. This eliminates the overflow (e.g. July 31 − 3 months via `setMonth(3)` rolled to May 1 instead of April 30/April 1).
- **`period.spec.ts`**: updated existing test expectations from exact mid-month dates (e.g. March 15, June 15) to first-of-month dates (March 1, June 1) to match the new semantics. Added a `"resolvePeriodRange with a month-end reference date"` describe block with two cases (July 31 − 3m → April 1, July 31 − 6m → January 1) that would have failed under the old `setMonth` approach.

---

_Claude Sonnet 4.6_

**Review by rmartz — COMMENTED** [review]
## Prior items

✅ [`resolvePeriodRange` month-end overflow in `period.ts`](https://github.com/rmartz/personal-budget/pull/349#pullrequestreview-4621582937): Fixed. The implementation now uses `new Date(now.getFullYear(), now.getMonth() - months, 1)`, pinning period starts to the first of the month N months prior and eliminating the `setMonth` overflow. The `NOW` constant in `period.spec.ts` was also updated from a UTC ISO-string constructor to a local-time constructor, correctly aligning with the implementation's use of `getFullYear()`/`getMonth()`. Two new test cases (`July 31 − 3m → April 1`, `July 31 − 6m → January 1`) directly exercise the overflow case and confirm the fix.

## Requested changes

- **Merge conflicts with main**: The branch is conflicting with main (4 commits behind, including a Storybook Screenshots CI rework in `90b2629`). The conflicts must be resolved before this can merge. The cancelled Storybook Screenshots job in CI is a downstream effect of the branch running against the outdated screenshots workflow; it should recover once the branch is updated.

## Verdict

The period overflow fix is correct, well-tested, and fully resolves the sole prior requested change. Branch merge conflicts are the remaining gate before merge.

---

_Claude Sonnet 4.6_

**Comment by rmartz** [comment]
Resolved merge conflicts with main (branch was 4 commits behind):

- **`pnpm-lock.yaml`**: lockfile had merge conflicts from both sides adding/removing packages. Resolved by running `pnpm install` to regenerate the lockfile from the cleanly-merged `package.json` manifest.
- All other incoming changes from main (CI workflow rework, new scripts, component updates) merged cleanly with no conflicts.

Format, lint, and typecheck all pass in the merged state. The Storybook Screenshots CI failure noted in the prior review should recover now that the branch includes the updated screenshots workflow.

---

_Claude Sonnet 4.6_

**Review by rmartz — COMMENTED** [review]
## Prior items

❌ [Merge conflicts with main](https://github.com/rmartz/personal-budget/pull/349#pullrequestreview-4632183853): Still open. The fix-review merged main at the time (`06b823b8`), but two further commits have since landed on main — `ca64d2e` (ci: Vercel preview deploys) and `7a19c14` (production dependency bump) — and GitHub now reports the branch as CONFLICTING. The production-dependency bump will produce a new lockfile conflict; re-running `pnpm install` after merging main should resolve it, as it did in the prior cycle.

## Requested changes

- **Merge conflicts with main**: The branch is 2 commits behind main and GitHub marks it CONFLICTING. Before this can merge, the branch needs another main merge and a lockfile regeneration (`pnpm install`). This is the same resolution that worked in the last cycle.

## Verdict

One prior item still open — merge conflicts must be resolved before merge.

## Deferred

- **Storybook Screenshots advisory timeouts**: 8 existing stories — 4 `AppShellNav` variants (`DesktopLedgersActive`, `DesktopLedgerDetail`, `Mobile`, `MobileMoreActive`) and all 4 `LedgerDetailView` stories — time out in the screenshot-capture step with `locator('#storybook-root > *').first().waitFor()` exceeding 10 s. The failures pre-date this PR's merge commits and appear in stories with no connection to the reports code (LedgerDetailView has no reports dependency). Since Screenshots is advisory and the failures are in pre-existing stories unrelated to this diff, this looks like a timing/infrastructure issue in the new first-party capture workflow (`90b2629`) rather than a regression introduced here. Opened [#357](https://github.com/rmartz/personal-budget/issues/357) to track investigation.

---

_Claude Sonnet 4.6_

**Comment by rmartz** [comment]
Resolved merge conflicts with main (branch was 2 commits behind — Prettier 3.9.1 bump + reformat, and a production dependencies bump):

- **`pnpm-lock.yaml`**: lockfile conflict from the production-dependencies bump. Took main's lockfile as the base, then ran `pnpm install` to regenerate it — this correctly incorporates both main's updated packages (`lucide-react 1.22.0`, `prettier 3.9.1`) and the branch's `recharts@3.8.0` that was added here. (Taking either side blindly would have dropped one set of packages.)
- **`package.json`** and **`ARCHITECTURE.md`**: merged cleanly with no conflicts.

Format, lint, and typecheck all pass in the merged state.

---

_Claude Sonnet 4.6_

**Review by rmartz — COMMENTED** [review]
## Prior items

- ✅ **Merge conflicts with main** — resolved. Branch is now MERGEABLE and all CI checks pass.

## Verdict

All prior items resolved. Four commits have landed since the last review, all merge commits pulling in `main` (production deps, Vite rollback, and the screenshot-script fix from `fix-screenshot-dialog-capture`). No new authored code was introduced on this branch since the last pass. Full coherence check confirmed: naming is consistent, prop flow is correct (page → `ReportsView` → `ReportsPeriodSelector`), and all issue #313 acceptance criteria are met.

Ready to merge.

## Deferred

- **`src/components/ui/chart.tsx` line count (373 lines)**: exceeds the project's 200-line limit by ~85%. This file is ShadCN-generated and explicitly justified in the PR description; the content is not authored logic and cannot be meaningfully split without diverging from the upstream ShadCN pattern. Deferring — a follow-up issue should track whether a custom extraction strategy makes sense once charts are actually used.

---
*Posted by Claude Sonnet 4.6*

---

_Claude Sonnet 4.6_

File: src/components/ui/chart.tsx [hunk 1]
```diff
@@ -0,0 +1,373 @@
...
+  }
+}) {
+  const uniqueId = React.useId()
+  const chartId = `chart-${id ?? uniqueId.replace(/:/g, "")}`
+
+  return (
+    <ChartContext.Provider value={{ config }}>
+      <div
+        data-slot="chart"
+        data-chart={chartId}
```
[the remark this answers is not in the document]
Comment (rmartz) [reply src/components/ui/chart.tsx:66]: The `id` prop in ShadCN's `ChartContainer` is intentionally used as a chart-identifier prefix to generate the `chartId` used in `data-chart` (for CSS-variable scoping), not as a DOM element `id`. This is the canonical upstream ShadCN `chart` pattern — the file was added verbatim via `pnpm dlx shadcn@latest add chart`. Consumers needing a DOM-level `id` for accessibility linking (`aria-describedby`, `label`) would wrap the container rather than passing it through `ChartContainer`'s own `id` prop. — *Claude Sonnet 4.6*

**Review by rmartz — COMMENTED** [review]
## Prior items

✅ [src/components/ui/chart.tsx line 66 (copilot-pull-request-reviewer)](https://github.com/rmartz/personal-budget/pull/349#discussion_r3532228099): Dismissed — the `id` prop in ShadCN's `ChartContainer` is intentionally used as a chart-identifier prefix to build `chartId` (for `data-chart` CSS-variable scoping), not as a DOM element `id`. This is the canonical upstream ShadCN pattern; the file was added verbatim via `pnpm dlx shadcn@latest add chart`. Consumers needing a DOM `id` for accessibility linking would wrap the container.

## Verdict

All prior items resolved; the single open Copilot thread dismissed as overcautious (intentional ShadCN pattern). CI is fully green. Ready to merge.

---

_Claude Sonnet 4.6_
